### PR TITLE
Combine GlobalFlags and EngineFlags into EngineOptions

### PR DESCRIPTION
**Pull request by jwhonce**
* EngineOptions obtained in command via
  `opts, err := registry.Options(cmd)`

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podmanV2/containers/wait.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func init() {
 	flags.DurationVarP(&waitFlags.Interval, "interval", "i", time.Duration(250), "Milliseconds to wait before polling for completion")
 	flags.BoolVarP(&waitFlags.Latest, "latest", "l", false, "Act on the latest container podman is aware of")
 	flags.StringVar(&waitCondition, "condition", "stopped", "Condition to wait on")
-	if registry.EngineOpts.EngineMode == entities.ABIMode {
+	if registry.EngineFacts.EngineMode == entities.ABIMode {
```
Comment (mheon) [inline cmd/podmanV2/containers/wait.go]: I really dislike this name. What was wrong with EngineOpts?

Comment (jwhonce) [reply cmd/podmanV2/containers/wait.go]: Since I was combining to structs, I came up with a 3rd name.  I can use EngineOpts.  No strong opinion.

Comment (rhatdan) [reply cmd/podmanV2/containers/wait.go]: SGTM,

**Comment by TomSweeneyRedHat** [comment]
LGTM other than the name discussion.

**Comment by rhatdan** [comment]
@jwhonce Needs rebase.

**Pull request merged**